contrib/ros/envs/sawyer_env.py

```diff
@@ -0,0 +1,346 @@
+                When the distance from robot endpoint to the target point is smaller than this threshold
+                the training episode is done.
+                To calculate the sparse reward, if the distance is smaller than this threshold returns -1
+                else returns 0.
+        :param initial_model_pos: dict
+                joint positions and object positions
+        :param sparse_reward: bool
+                if reward_type is 'sparse'
+        :param target_in_the_air: bool
+                set to see if the target can be in the air
+        :param simulated: bool
+                if the environment is for real robot or simulation
+        :param target_range: float
+                the new target position for the new training episode would be randomly sampled inside
+                [initial_gripper_pos - target_range, initial_gripper_pos + target_range]
+        :param obj_range: float
+                the new initial object x,y position for the new training episode would be randomly sampled inside
+                [initial_gripper_pos - obj_range, initial_gripper_pos + obj_range]
+        """
+        # run the superclass initialize function first
+        Serializable.quick_init(self, locals())
+
+        RosEnv.__init__(self)
+        # long-running ROS/robot setup
+        self.initialize()
+
+        # Verify robot is enabled
+        self._robot = Sawyer(
+            initial_robot_joint_pos, control_mode=robot_control_mode)
+        if not self._robot.enabled:
+            raise RuntimeError(
+                '[Error] SawyerEnv.__init__: The robot is not enabled!')
+
+        self._simulated = simulated
+        if self._simulated:
+            rospy.Subscriber('/gazebo/model_states', ModelStates,
+                             self.update_model_states)
+            self._gazebosrv = Gazebo()
+        else:
+            # TODO(gh/1: Add vicon system utility for models' states update)
+            pass
+        self._initial_model_pos = initial_model_pos
+        self.has_object = has_object
+        self.distance_threshold = distance_threshold
+        self.sparse_reward = sparse_reward
+        self.target_range = target_range
+        self.target_in_the_air = target_in_the_air
+        self.obj_range = obj_range
+
+        rospy.on_shutdown(self._shutdown)
+
+        self.initial_setup(self._initial_model_pos)
+
+    def _shutdown(self):
+        logger.log('Exiting...')
+        if self._simulated:
+            # delete model
+            logger.log('Delete gazebo models...')
+            self._gazebosrv.delete_gazebo_model(model_name='cafe_table')
+            self._gazebosrv.delete_gazebo_model(model_name='block')
+            self._gazebosrv.delete_gazebo_model(model_name='target')
+
+    def update_model_states(self, data):
+        if self._simulated:
+            self.model_states = data
+        else:
+            # TODO(gh/2: Finish the 'update_model_states' function for real robot)
```

Comment:
```python
# TODO(gh/???): Finish the 'update_model_states' function for real robot)
```
Where `???` is the issue number github assigns your todo when you file it here:https://github.com/ryanjulian/rllab/issues

Reply:
Got it I am using gh/8 Sawyer runtime support